jupyter_server/extension/application.py

```diff
@@ -140,6 +140,38 @@ def static_url_prefix(self):
         help=_("The default URL to redirect to from `/`")
     )
 
+    custom_display_url = Unicode(u'', config=True,
+        help=_("""Override URL shown to users.
+
+        Replace actual URL, including protocol, address, port and base URL,
+        with the given value when displaying URL to the users. Do not change
+        the actual connection URL. If authentication token is enabled, the
+        token is added to the custom URL automatically.
+
+        This option is intended to be used when the URL to display to the user
+        cannot be determined reliably by the Jupyter server (proxified
+        or containerized setups for example).""")
+    )
+
+    @default('custom_display_url') 
+    def _default_custom_display_url(self):
+        """URL to display to the user."""
+        # Get url from server.
+        url = url_path_join(self.serverapp.base_url, self.default_url)
+        return self.serverapp.get_url(self.serverapp.ip, url)
```

Comment:
Won't this fail if `self.token` is truthy?

Reply:
:+1: fixing

Reply:
In #115, this returns a url without a token. That's what we want here. The `display_url` property adds a token to `custom_display_url`.